### ISSUE TITLE
equivalence handling in prerequisites

### ISSUE THREAD

**Issue by paradite** [bsmmoon/KBMP#27]
we need to handle modules' equivalence in prerequisites

we can do the same style like preclusion #22 :

```
(equivalence "CS1010" "CS1010E" "CS1010R" "CS1010J" "CS1101S" "CS1010X")
```

or we can add the info directly to the module template.
1. find equivalences of module A
2. match a module B planned/taken
3. test if the module B is one of the equivalence

Both incur some extra time in pattern matching, not sure which one is preferred.

**Comment by paradite** [comment]
@bsmmoon

**Comment by bsmmoon** [comment]
It seems there is no `equivalence` field in the `module` object. @joey-c How is the equivalence represented?

**Comment by bsmmoon** [comment]
blocked by #37

**Comment by paradite** [comment]
implemented 5578c06da894412b889fd997c0e5e2ad68d80610
